### DISCUSSION TITLE
Runtime-Agnostic Async support

### DISCUSSION THREAD

**Discussion by jtsylve** [oxigraph/oxigraph#1122] [Q&A]
Is there currently anything tokio-specific with the async parsers and serializers or could they be made executor agnostic by using shared traits from futures_io?

**Comment by Tpt** [comment]
Hi! This is a good point. The existance of two different traits is sad. I have implemented the Tokio traits because its what I mostly care about but nothing in the implementation is specific to Tokio. Two possible ways of working with futures_io:
- Use [`tokio-utils` `FuturesAsyncReadCompatExt`](https://docs.rs/tokio-util/latest/tokio_util/compat/trait.FuturesAsyncReadCompatExt.html) to wrap a `futures_io::AsyncRead` into a `tokio::AsyncRead`
- Copy paste the Tokio-related code to use `futures_io::AsyncRead` instead. I guess only minor changes will be needed. Contribution welcome!
